### DISCUSSION TITLE
IO Error: Can't read `blocks/index/LOCK` while Bitcoin Core is running

### DISCUSSION THREAD

**Discussion by nym21** [Congyuwang/Rusty-Bitcoin-Explorer#27] [General]
Hi !

I wanted to try your crate as it might be exactly what I'm looking for (so thank you a lot for your work !) but this example won't run while `Bitcoin Core` (24.0.1) is running, which is slightly inconvenient.

```rust
fn main() {
    let path = Path::new("/Volumes/t7s/bitcoin");

    let db = BitcoinDB::new(path, false);

    let block_count = db.get_block_count();

    let total_number_of_transactions = (0..block_count)
        .map(|i| db.get_header(i).unwrap().n_tx)
        .sum::<u32>();
}
```

Everything is fine when I kill, it's weird that I seem to be the first one to notice this issue though ?

I have the blocking data on an external SSD and am using an M1 Mac (13.3.1) if that helps.

I hope there is an easy fix !

**Comment by Congyuwang** [comment]
Hi. Bitcoin core and this crate can’t run at the same time. To read the block files, Bitcoin core needs to be stopped. That’s a current limitation by design though.

**Comment by nym21** [comment]
Oh, that's unfortunate.
I thought it would work like the `blocks_iterator` crate.

Thank you for your quick answer though
